src/task.c

```diff
@@ -0,0 +1,32 @@
+#include <stdint.h>
+
+#include "task.h"
+#include "asm.h"
+#include "malloc.h"
+
+static unsigned int *tasks[TASK_LIMIT];
+
+void    task_add(void (*task_code)(void), uint16_t stack_size)
+{
+    int i = 0;
+
+    while (tasks[i])
+        i++;
+    tasks[i] = (unsigned int*)malloc(sizeof(unsigned int) * stack_size);
+
+    tasks[i] += stack_size - 8;
```

Comment:
Where 8 is coming from?

Reply:
During context switch we push r0-r7 there, thus we need space for them. Still, I think it will be good to add a define. Any ideas?